clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -3869,19 +3951,60 @@ private List<TopicPartition> getPartitionsFromV3Request(AddPartitionsToTxnReques
         return AddPartitionsToTxnRequest.getPartitions(request.data().v3AndBelowTopics());
     }
 
-    private void prepareEndTxnResponse(Errors error, final TransactionResult result, final long producerId, final short epoch) {
-        this.prepareEndTxnResponse(error, result, producerId, epoch, false);
-    }
+    private void prepareEndTxnResponse(
```

Comment:
Can we maybe put a comment on these methods about what should be used when and/or find a way to combine them?